File: 2022/01/202201.lua

-- 'pure' lua version
-- function part1()
--     local calories = {0}
--     local ei = 1
--     local mxi = 1
--     for line in io.lines("input.txt") do
--         if line == "" then
--             calories[#calories+1] = 0
--             ei = ei + 1
--         else
--             local n = tonumber(line)
--             calories[ei] = calories[ei] + n
--             if calories[ei] > calories[mxi] then
--                 mxi = ei
--             end
--         end
--     end
--     return calories[mxi]
-- end

require 'pl.strict'
local seq = require 'pl.seq'
require('pl.stringx').import()

local input = io.input("input.txt"):read('*a')

local elves = input:split("\n\n"):map(function (group)
    return seq.sum(group:splitlines():map(tonumber))
end)

local _, ans1 = seq.minmax(elves)
print("PART1:", ans1)

elves:sort()
local ans2 = seq.sum(elves:slice(-3,-1))
print("PART2:", ans2)

File: 2022/01/202201.py
def read_data(filename):
  with open(filename) as f:
    return [ [int(line)
      for line in group.splitlines()]
        for group in f.read().split('\n\n')]

tinput = read_data("tinput.txt")
input = read_data("input.txt")

def part1(data):
  return max(sum(x) for x in data)

assert part1(tinput) == 24000
print("PART1", part1(input))

def part2(data):
  return sum(list(sorted(sum(x) for x in data))[-3:])

assert part2(tinput) == 45000
print("PART2", part2(input))
